src/pygeoprocessing/routing/routing.pyx

```diff
@@ -468,9 +468,17 @@ cdef class _ManagedRaster:
 
         raster_band = None
         if self.write_mode:
-            raster = gdal.OpenEx(
-                self.raster_path, gdal.GA_Update | gdal.OF_RASTER)
-            raster_band = raster.GetRasterBand(self.band_id)
+            max_retries = 5
+            while max_retries > 0:
+                raster = gdal.OpenEx(
+                    self.raster_path, gdal.GA_Update | gdal.OF_RASTER)
+                if raster is None:
+                    max_retries -= 1
+                    LOGGER.error(
+                        f'unable to open {self.raster_path}, retrying...')
+                    time.sleep(0.2)
+                    continue
+                break

```

Comment:
I'm not up to speed on this set of code, but if raster is NOT `None`, do we need to get the `raster_band`, and if it times out without returning `None`, should there be an error or is it find to just it continue? Perhaps I'll answer my own questions here as I read through.

Reply:
oh woof thanks for catching that.. that'd be a nightmare if that got out. patched in 1fe4d3f5dfbf90db2529ee4693a43c7a817e8e5b

this wouldn't show up unless the raster were very big, so it's hard to make test cases of this size since we don't want to wait 15 minutes for  test to run. stuff like this though makes me wary.